ipaserver/secrets/kem.py

```diff
@@ -130,17 +130,16 @@ def set_key(self, usage, principal, key):
         service_rdn = ('cn', servicename) if servicename != 'host' else DN()
         dn = str(DN(('cn', name), service_rdn, self.keysbase))
         try:
-            mods = [('objectClass', ['nsContainer',
-                                     'ipaKeyPolicy',
-                                     'ipaPublicKeyObject',
-                                     'groupOfPrincipals']),
-                    ('cn', name),
-                    ('ipaKeyUsage', RFC5280_USAGE_MAP[usage]),
-                    ('memberPrincipal', principal),
+            mods = [('objectClass', [b'nsContainer',
+                                     b'ipaKeyPolicy',
+                                     b'ipaPublicKeyObject',
+                                     b'groupOfPrincipals']),
+                    ('cn', name.encode('utf-8')),
+                    ('ipaKeyUsage', RFC5280_USAGE_MAP[usage].encode('utf-8')),
+                    ('memberPrincipal', principal.encode('utf-8')),
```

Comment:
How hard would it be to refactor this to use `ipaldap` (which handles encoding for you)?